### PR TITLE
Fix string literal project name in the DPA V2 task

### PR DESCRIPTION
**Pull request by jahsu-MSFT**
https://dev.azure.com/mseng/AzureDevOps/_workitems/edit/1651619

### CODE REVIEW

File: src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs [hunk 1]
```diff
@@ -316,5 +323,24 @@ private BuildResult GetResultFilter(bool allowPartiallySucceededBuilds, bool all
...
+
+            TeamProject proj = null;
+
+            try
+            {
+                proj = await projectClient.GetProject(projectName);
+            }
+            catch (Exception ex)
+            {
+                throw new ArgumentException("Get project failed" + projectName + " , exception: " + ex);
```
Comment (johnterickson) [inline src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs]: need a space

File: src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs [hunk 2]
```diff
@@ -316,5 +323,24 @@ private BuildResult GetResultFilter(bool allowPartiallySucceededBuilds, bool all
 
             return result;
         }
+      
+        private async Task<Guid> GetProjectId(AgentTaskPluginExecutionContext context, string projectName)
```
Comment (johnterickson) [inline src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs]: GetProjectIdAsync

File: src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs [hunk 3]
```diff
@@ -173,7 +175,12 @@ public class DownloadPipelineArtifactTaskV2_0_0 : PipelineArtifactTaskPluginBase
                 {
                     throw new ArgumentNullException("Project Name cannot be null.");
                 }
-                Guid projectId = Guid.Parse(projectName);
```
Comment (srdorbal) [inline src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs]: Was this variable (projectId) not used before?

File: src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs [hunk 4]
```diff
@@ -173,7 +175,12 @@ public class DownloadPipelineArtifactTaskV2_0_0 : PipelineArtifactTaskPluginBase
...
+                Guid projectId; 
+                bool isProjGuid = Guid.TryParse(projectName, out projectId);
+                if(!isProjGuid)
```
Comment (srdorbal) [inline src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs]: space

File: src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs (hunk 3 above)
Comment (jahsu-MSFT) [reply src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs]: It is.  I recreated one that will be assigned by the TryParse() function instead.

**Pull request merged**